### ISSUE TITLE
Possible mistake in "wheretheressmoke" of subject S01. 

### ISSUE THREAD

**Issue by dyhan316** [HuthLab/encoding-model-scaling-laws#4]
Hello again! The response data for 'wheretheressmoke" of subject S01 seems to be averaged in an incorrect way. 

In other words when I calculate `tensession_resp.mean(axis=0)/avg_resp_dict['wheretheressmoke']`, the values are not 1 for sub1, while they are 1 for sub2 and 3. 

```
#Sub1 
array([[ -0.5260241 ,   0.43809538,   0.72644223, ...,  -0.12333434,
         -0.06862524,  -0.17878285],
       [  0.22232635, -66.78132633,  -0.44553775, ...,  -0.74112664,
         -0.9559144 ,   0.36624247],
       [  0.75621392,   0.79440582,   0.69228175, ...,  -1.66421863,
         -1.03083738,  -0.93969021],
       ...,
       [  1.98471561,   0.2910771 ,   0.55035473, ...,  -0.44772529,
          1.72803771,  -0.8670477 ],
       [ -0.14013489,  -0.90228549,  -0.38435461, ...,   0.51850835,
         -0.40164749,   0.51553115],
       [  0.16714859,   0.11103255,  -0.30769317, ...,   0.51741804,
          1.05856689,   0.30922208]])
 #Sub2 
array([[1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       ...,
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.]])

#Sub3 
array([[1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       ...,
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.]])

```

I found that this may be due to the fact that the Sub1's wheretheressmoke data was normalized after adding them up, as you can see in the average and std values below 

```
#Sub1
itsabox ((355, 81126), 4.7093635045150747e-17, 0.9999999999999991)
odetostepfather ((404, 81126), -4.183271636501786e-18, 1.0)
inamoment ((205, 81126), -3.715829144698577e-17, 1.0000000000000004)
wheretheressmoke ((291, 81126), -3.1167046066065954e-17, 1.0)
onapproachtopluto ((271, 81126), -5.8329725420055375e-18, 0.5123625396215089)
fromboyhoodtofatherhood ((348, 81126), -4.8967107880959505e-18, 0.43156861943485947)

#Sub2
itsabox ((355, 94251), 3.499718323405976e-17, 1.0000000000000004)
odetostepfather ((404, 94251), -3.9085500487734813e-17, 1.0000000000000016)
inamoment ((205, 94251), -8.414075334487538e-18, 0.999999999999999)
wheretheressmoke ((291, 94251), -1.8649678783029527e-17, 0.3592392686995733)
onapproachtopluto ((271, 94251), 2.079494437168422e-17, 0.4883584127554785)
fromboyhoodtofatherhood ((348, 94251), 4.055373299929339e-17, 0.4661844447316095)

#Sub3 
itsabox ((355, 95556), 1.9632392991683566e-17, 0.9999999999999999)
odetostepfather ((404, 95556), 3.3595442989632425e-17, 0.9999999999999987)
inamoment ((205, 95556), 2.608143194392676e-17, 1.0000000000000013)
wheretheressmoke ((291, 95556), 3.090872077370644e-18, 0.36932671494961317)
onapproachtopluto ((271, 95556), -2.6657210209366712e-17, 0.4865749028668901)
fromboyhoodtofatherhood ((348, 95556), -3.145461764665795e-17, 0.4530105777188318)

```

the code for  getting the things above are attached below for reference : 

```
#checking some stuff....
import os 
import numpy as np 
import joblib

story2see = ['itsabox', 'odetostepfather', 'inamoment', 'wheretheressmoke', 'onapproachtopluto', 'fromboyhoodtofatherhood']
os.chdir('/scratch/x2767a03/BRAIN_DECODING/brain_scaling_law/sample_data/wheretheressmoke_preds')

sub = "S01" #S2,S3 : works 

avg_resp_dict = joblib.load(f'UT{sub}_responses.jbl')
tensession_resp = joblib.load(f'tensessions_wheretheressmoke_{sub}.jbl')
os.listdir()


stories = avg_resp_dict.keys()


def get_stats(arr):
    arr = np.nan_to_num(arr)
    return arr.shape, np.mean(arr), np.std(arr) #, np.min(arr), np.max(arr)

for story in story2see : 
    print(story, get_stats(avg_resp_dict[story]))
    
    
for trial in range(10):
    print(f'trial {trial}', get_stats(tensession_resp[trial]))
    
print(tensession_resp.mean(axis=0)/avg_resp_dict['wheretheressmoke'])


from sklearn.metrics import r2_score

print(r2_score(tensession_resp.mean(axis=0), avg_resp_dict['wheretheressmoke']), avg_resp_dict['wheretheressmoke'].std(axis=0))


```

**Comment by RAntonello** [comment]
Hi Danny,

Thanks for pointing this out. Fortunately, I do not think it matters terribly much, as all of our metrics are correlation-based and therefore scale invariant. I will make a note to fix it. As an aside, I noticed that you are working on some decoding stuff, perhaps trying to apply these larger models for semantic decoding. I would recommend you read this new preprint by our lab that may be doing something similar: https://arxiv.org/abs/2405.14055. Furthermore, if you are planning to use this dataset for decoding, you should avoid using averages at all (we tend to use the first instance of each story).  Additionally, I want to reiterate that the metric we used in this work is r^2, not R^2 (there is a difference). 

Thanks,

RJ

**Comment by dyhan316** [comment]
Thank you for your detailed response! Actually as of now I am only trying to improve the encoding model and recreating the results of your paper, but thank you for the paper recommendation! 

Oh and regarding the distinction between R^2 and r^2, I use `np.corrcoef` over each voxel like in your code when calculating `corrs_unnorm` (code is attached below )

I have one question if you don't mind : during ridge regression, was the bootstrap number and alpha candidates and chunk len the same as the values you used in the `ipynb` notebook example? (like below)(just wanted to make sure!)

```
alphas = np.logspace(1, 4, 15) # Equally log-spaced ridge parameters between 10 and 10000. 
nboots = 3 # Number of cross-validation ridge regression runs. You can lower this number to increase speed.

# Bootstrap chunking parameters
chunklen = 20
nchunks = int(len(Rresp) * 0.25 / chunklen)

# Run ridge regression - this might take some time
wt, corr, alphas, bscorrs, valinds = bootstrap_ridge(delRstim, Rresp, delPstim, Presp,
                                                     alphas, nboots, chunklen, nchunks,
                                                     use_corr=False, single_alpha=False)
```


Code for calculating r vector (then I did `np.abs(r)*r` to get the resulting r^2 value as the paper said) :) 
```
def compute_r_single_trial(single_trial_data, data_pred):    
    """ 
    input 
    * single_trial_data : 2D numpy array (timepoints, voxels)
    * data_pred : 2D numpy array (timepoints, voxels)
    
    output
    * vox_wise_corr : 1D numpy array (voxels) (correlation coefficient for each voxel)
    #voxel flatten해서 한다던지 하면 안됨!!
    """
    vox_wise_corr = np.zeros(single_trial_data.shape[1])
    for vox in range(single_trial_data.shape[1]):
        vox_wise_corr[vox] = np.corrcoef(single_trial_data[:,vox], data_pred[:,vox])[0][1]
    return vox_wise_corr    
```

Thank you again!

Danny

**Comment by kroq-gar78** [comment]
> during ridge regression, was the bootstrap number and alpha candidates and chunk len the same as the values you used in the `ipynb` notebook example? (like below)(just wanted to make sure!)
> 
> ```
> alphas = np.logspace(1, 4, 15) # Equally log-spaced ridge parameters between 10 and 10000. 
> nboots = 3 # Number of cross-validation ridge regression runs. You can lower this number to increase speed.
> 
> # Bootstrap chunking parameters
> chunklen = 20
> nchunks = int(len(Rresp) * 0.25 / chunklen)

Yes, those are the parameters we used to fit all encoding models.
